src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/PTMendRelationSupportClasses.java

```diff
@@ -0,0 +1,447 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.openstreetmap.josm.tools.I18n.tr;
+
+public class PTMendRelationSupportClasses {
+
+    public static class PTMendRelationPaintVisitor extends PaintVisitor {
+        protected final String I18N_ADD_ONEWAY_VEHICLE_NO_TWO_WAY;
+        protected final String I18N_CLOSE_OPTIONS = I18n.marktr("Close the options");
+        protected final String I18N_NOT_REMOVE_WAYS = I18n.marktr("Do not remove ways");
+        protected final String I18N_REMOVE_CURRENT_EDGE = I18n.marktr("Remove current edge (white)");
+        protected final String I18N_REMOVE_WAYS = I18n.marktr("Remove ways");
+        protected final String I18N_REMOVE_WAYS_WITH_PREVIOUS_WAY = I18n.marktr("Remove ways along with previous way");
+        protected final String I18N_SKIP = I18n.marktr("Skip");
+        protected final String I18N_SOLUTIONS_BASED_ON_OTHER_RELATIONS = I18n.marktr("solutions based on other route relations");
+        protected final String I18N_TURN_BY_TURN_NEXT_INTERSECTION = I18n.marktr("turn-by-turn at next intersection");
+        protected final String I18N_BACKTRACK_WHITE_EDGE = I18n.marktr("Split white edge");
+
+        protected final Color[] FIVE_COLOR_PALETTE = {
+            new Color(0, 255, 0, 150),
+            new Color(255, 0, 0, 150),
+            new Color(0, 0, 255, 150),
+            new Color(255, 255, 0, 150),
+            new Color(0, 255, 255, 150)
+        };
+
+        protected final Color CURRENT_WAY_COLOR = new Color(255, 255, 255, 190);
+        protected final Color NEXT_WAY_COLOR = new Color(169, 169, 169, 210);
+
+        protected static final Map<Character, Color> CHARACTER_COLOR_MAP = new HashMap<>();
+        static {
+            CHARACTER_COLOR_MAP.put('A', new Color(0, 255, 0, 200));
+            CHARACTER_COLOR_MAP.put('B', new Color(255, 0, 0, 200));
+            CHARACTER_COLOR_MAP.put('C', new Color(0, 0, 255, 200));
+            CHARACTER_COLOR_MAP.put('D', new Color(255, 255, 0, 200));
+            CHARACTER_COLOR_MAP.put('E', new Color(0, 255, 255, 200));
+            CHARACTER_COLOR_MAP.put('1', new Color(0, 255, 0, 200));
+            CHARACTER_COLOR_MAP.put('2', new Color(255, 0, 0, 200));
+            CHARACTER_COLOR_MAP.put('3', new Color(0, 0, 255, 200));
+            CHARACTER_COLOR_MAP.put('4', new Color(255, 255, 0, 200));
+            CHARACTER_COLOR_MAP.put('5', new Color(0, 255, 255, 200));
+        }
+
+
+        /** The graphics */
+        private final Graphics g;
+        /** The MapView */
+        private final MapView mv;
+        private HashMap<Way, java.util.List<Character>> waysColoring;
+
+        /**
+         *
+         * @param g
+         * @param mv
+         * @param PTorNot
+         */
+        public PTMendRelationPaintVisitor(Graphics2D g, MapView mv, boolean PTorNot) {
+            super(g, mv);
+            this.g = g;
+            this.mv = mv;
+
+            if (PTorNot) {
```

Comment:
Just a small thing about boolean variables that I noticed here:
Words like "or" and "not" that have a meaning in boolean logic should be avoided when naming boolean variables. It's often helpful to start the variable name with "has" or "is" to make clearer what is meant.

For this line here it is essentially like asking someone "Is this action for public transport or not?" and you answer "Yes" or "No". But responding with Yes/no to an or-Question isn't allowed.

I find avoiding "not" often useful for readability:`isActive` is easier to understand than `isNotActive`, especially when it is negated somewhere and become a double-negation (`if (!isNotActive && isNotSelected)` vs `if (isActive && !isSelected)`)

You could name it for example `isMendingPtRelation`, `isPT`, `forPT`, `isForPT` or similar.